### PR TITLE
[C#] Third party rules 

### PR DESCRIPTION
**Pull request by karan-batavia**

### CODE REVIEW

File: rules/sinks/third_parties/sdk/datadoghq/csharp.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+#  Sink rule for ThirdParty SDK
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Datadoghq.Www
+    name: Datadoghq Www
+    domains:
+      - "www.datadoghq.com/"
```
Comment (dbMundada) [inline rules/sinks/third_parties/sdk/datadoghq/csharp.yaml:10]: Update domain to `datadoghq.com`

File: rules/sinks/third_parties/sdk/hangfire/csharp.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+
+#  Sink rule for ThirdParty SDK
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Hangfire.Www
+    name: Hangfire Www
+    domains:
+      - "www.hangfire.io/"
```
Comment (dbMundada) [inline rules/sinks/third_parties/sdk/hangfire/csharp.yaml:10]: Update domain to`hangfire.io`

File: rules/sinks/third_parties/sdk/okta/csharp.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+
+#  Sink rule for ThirdParty SDK
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Okta.Www
+    name: Okta Www
```
Comment (dbMundada) [inline rules/sinks/third_parties/sdk/okta/csharp.yaml:8]: Name can be simple `Okta`
Domain `okta.com`

File: rules/sinks/third_parties/sdk/seven/csharp.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+#  The id follows a format : "ThirdParties.SDK.<THIRD_PARTY_ORGANISATION>.<SUB_ORGANISATION_IF_APPLICABLE>"
+
+sinks:
+
+  - id: ThirdParties.SDK.Seven
+    name: Seven
+    domains:
+      - "seven.io"
+    patterns:
+      - "(?i)(sms77-api).*"
```
Comment (dbMundada) [inline rules/sinks/third_parties/sdk/seven/csharp.yaml:12]: Is this pattern correct?

Comment (karan-batavia) [reply rules/sinks/third_parties/sdk/seven/csharp.yaml:12]: Yes.
Refer: https://www.nuget.org/packages/sms77-api/

Comment (dbMundada) [reply rules/sinks/third_parties/sdk/seven/csharp.yaml:12]:
```
using System;
using System.Threading.Tasks;
using Client = Sms77.Api.Client;

class Program
{
    static async Task Main()
    {
        var apiKey = Environment.GetEnvironmentVariable("SMS77_API_KEY");
        var client = new Client(apiKey);
        var balance = await client.Balance();
        Console.WriteLine($"Current account balance: {balance}");
    }
}
```

https://github.com/seven-io/net-client

I think pattern should be `(?i)(Sms77.Api.Client).*`

Comment (karan-batavia) [reply rules/sinks/third_parties/sdk/seven/csharp.yaml:12]: I see, rare case where package and namespace aren't the same.

Comment (karan-batavia) [reply rules/sinks/third_parties/sdk/seven/csharp.yaml:12]: Good catch, let me fix.

**Comment by karan-batavia** [comment]
Reopened #432

**Pull request closed without merging**